### PR TITLE
feat(Validators): Reworked validation behavior.

### PR DESCRIPTION
**Pull request by Hyperkid123**
Complete rework of validation functions. It is heavily based upon redux-form-validators.

- New global settings for validation messages
- Out of the box react.intl support
- Supporting many currently non existing validation

### CODE REVIEW

File: src/constants/index.js [hunk 1]
```diff
@@ -18,9 +18,17 @@ export const components = {
...
      */
   MIN_LENGTH: 'min-length-validator',
+  /**
+     * max length of the input value
+     */
+  MAX_LENGTH: 'max-length-validator',
+  /**
+   * exact length of input value
+   */
+  EXACT_LENGTH: 'exact-length',
```
Comment (rvsia) [inline src/constants/index.js]: Nitpick: other length validators have “validator” in name

File: src/validators/helpers.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import Validators from './validators';
+
+const HAS_PROP = {}.hasOwnProperty;
+export const TO_STRING = {}.toString;
+
+const isReactElement = object => typeof object === 'object' && object !== null && '$$typeof' in object;
```
Comment (rvsia) [inline src/validators/helpers.js]: What abou React.isValidElement() ?

File: src/validators/messages.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    id: 'form.errors.lessThan',
+    defaultMessage: number => `Value must be less than ${number}`,
+  },
+  lessThanOrEqualTo: {
+    id: 'form.errors.lessThanOrEqualTo',
+    defaultMessage: number => `Value must be less than or equal to ${number}`,
+  },
+  mustBeBool: {
+    id: 'forms.errors.mustBeBool',
+    defaultMessage: 'Value msut be boolean.',
```
Comment (rvsia) [inline src/validators/messages.js]: Must

File: src/tests/validators/validators.test.js [hunk 4]
```diff
@@ -1,93 +1,125 @@
...
-    it('should pass the validation', () => {
-      expect(minValue({ value: 99 })(123)).toBeUndefined();
+  describe('Length validator', () => {
+    it('should pass is 5 chars long validation', () => {
+      expect(validatorMapper(validators.EXACT_LENGTH)({ treshold: 5 })('12345')).toBeUndefined();
     });
 
-    it('should pass the validation if no input given', () => {
-      expect(minValue({ value: 99 })()).toBeUndefined();
+    it('should fail is 5 chars long validation', () => {
```
Comment (martinpovolny) [inline src/tests/validators/validators.test.js]: " fail is 5 chars " sounds wrong

File: src/tests/validators/validators.test.js [hunk 5]
```diff
@@ -1,93 +1,125 @@
...
+      expect(validatorMapper(validators.EXACT_LENGTH)({ treshold: 5 })('1234')).toBe('Should be 5 characters long.');
     });
 
-    it('should not pass the validation', () => {
-      expect(minValue({ value: 99 })(1)).toEqual('Should be greater or equal to: 99');
+    it('should fail is 5 chars long validation with custom message', () => {
```
Comment (martinpovolny) [inline src/tests/validators/validators.test.js]: and here

File: src/validators/helpers.js (hunk 2 above)
Comment (Hyperkid123) [reply src/validators/helpers.js]: Well that will restrict these functions only to React. I am not sure if that is something we want.

**Pull request merged**